### ISSUE TITLE
Documentation Issue

### ISSUE THREAD

**Issue by buteomont** [usatiuk/EggbotWireless#1]
Hi. This may not be the best place to ask about this; I apologize if so.
I've built the eggbot using a Wemos D1 R2, and have compiled and loaded your code into it.
I need to modify the port assignments to match my eggbot, but can't for the life of me figure out where to do it.  I'm usually pretty good at this stuff but this one has me stymied. 

Also, I don't see much in the way of documentation describing how to actually send the gcode to the eggbot.  I've figured out some stuff, such as how to get the status, using cURL and using PlatformIO's serial terminal. I can get the status in a web browser, but that's about all.  Any help you can offer will be appreciated.

Thanks for sharing this project! It's the only eggbot I can find that works on the ESP8266.

**Comment by usatiuk** [comment]
Hello, and thank you for your interest in this thing!

What ports exactly do you want to change? 
For stepper driver pins, you can change them in AVR's `Globals.cpp` - they're the first two arguments to Stepper class initializer.
Servo's pin is in `Pen.cpp` file, again in the initializer. (Now that I'm describing it, it really could be better)
I2C pins on the esp8266 you can change at `Wire.begin(SDA, SCL)` in `main.cpp` 

About sending gcode: going to the eggbot's root page should open an interface to upload gcode — a text box and a send button. Also I'm pretty sure you can just send the gcode commands via serial. At least that's how it supposed to work, but I can't really test anything because I don't have access to any hardware right now.

**Comment by buteomont** [comment]
Thank you for your response.  I think I figured out why I was so confused.  I was under the impression that this eggbot used _only_ an ESP8266 for its controller.  I see now that it still requires an Arduino for the motor control.
The eggbot that I am building uses a WEMOS D1 R2 board, and doesn't have any kind of AVR components at all.  That's been my problem with adapting Arduino firmware for this project all along; seems like they all have some sort of low-level hardware timer manipulation that doesn't jibe with the ESP8266.  I even tried paring down and using the Marlin 3D printer firmware, but even it won't compile for an ESP8266. 

After loading your code (the parts in the Firmware/EggbotWireless directory) I still couldn't get any response from the device in a web browser.  I could connect and ping it fine,  but could never get it to give me any kind of web page, other than the /getStatus page.

If I'm still misunderstanding things, please let me know.  I've been looking at so many different eggbot firmwares I may be conflating them!

(We can move this conversation to email if you want)

**Comment by usatiuk** [comment]
Yes, there's an additional AVR chip in the design that is used basically as a fancy I/O expander, which communicates with the ESP8266 via I2C (somewhat poorly though)

For the web interface you should also build and upload the web interface that is in the App folder. It's a nodejs app, so if you have nodejs set up I think `npm i` in `App` and then running the package script `buildweb.py` in `Firmware/EggbotWireless` should be enough. Then you have to build the image and upload it — you can do it with platformio, it should be in tasks menu in vscode, unfortunately I don't know the exact commands.

No, it's okay, I think it will be useful if anyone else stumbles across this project.

**Comment by buteomont** [comment]
Ok, that explains a lot.  Sounds a lot more complicated than what I was looking for.  I don't have Node.js installed (it's a bit of a hog) and have never fiddled with TypeScript, React, or any of that other stuff. I'm sure I could learn it but what I'm looking for is just some ESP8266 code that will accept gcode and write it on a golf ball.  Even if I got the web page working, I'd still have the AVR stuff to figure out. But hey, I learned about littleFS - I didn't know that SPIFFS had been replaced!

I'll probably just wind up writing something from scratch, or plagiarize bits of code from several eggbot projects and combine them.  It seems that every eggbot firmware out there is tied to the AVR processors at a low level in some way or another, so it looks like that's going to be new code.

Thanks for indulging me and explaining this.  You have an impressive project here!
